### PR TITLE
Refactor Client Code, part 7/7: scheduler

### PR DESCRIPTION
**Pull request by brandonforster**
This PR is the seventh part of several working on #196.

This PR refactors all of scheduler's client methods.

### CODE REVIEW

File: clients/scheduler/interval.go [hunk 1]
```diff
@@ -49,84 +46,100 @@ type IntervalClient interface {
...
-	return clients.UpdateRequest(s.url, interval, ctx)
+func (ic *intervalRestClient) Intervals(ctx context.Context) ([]models.Interval, error) {
+	return ic.requestIntervalSlice("", ctx)
 }
 
-//
-// Helper functions
-//
+func (ic *intervalRestClient) Update(interval models.Interval, ctx context.Context) error {
+	url, err := ic.urlClient.Prefix()
```
Comment (michaelestrin) [inline clients/scheduler/interval.go]: Variable naming in prior PRs (and elsewhere in this same file) is `urlPrefix`; rename `url` here for consistency.  Applies to any/all calls to Prefix().

Comment (brandonforster) [reply clients/scheduler/interval.go]: The distinction I made was that when there's no suffix added I just named it "url"

Comment (michaelestrin) [reply clients/scheduler/interval.go]: I see.  That wasn't immediately apparent to me.  Still prefer consistent naming of variable returned by Prefix(); `urlPrefix` makes the most sense to me.

**Comment by brandonforster** [comment]
Do not merge until #201 is merged.

**Pull request merged**